File: app/scripts/upload_resources.py
#!/usr/bin/env python3
import os
from supabase import create_client
from pathlib import Path

# Configuration Supabase (from upload_semestre3_to_supabase.py)
SUPABASE_URL = "https://gtnyqqstqfwvncnymptm.supabase.co"
SUPABASE_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Imd0bnlxcXN0cWZ3dm5jbnltcHRtIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2NTk4MjM1MCwiZXhwIjoyMDgxNTU4MzUwfQ.5iN2qsjSDvnHMOIx_iZMxiEnvW2SfzbGLDElv6Zun1A"

def upload_resource():
    supabase = create_client(SUPABASE_URL, SUPABASE_KEY)
    
    # Path relative to 'app' directory (assumed CWD)
    file_path = Path("../resources/competences/FREECAD_COURS.pdf")
    
    if not file_path.exists():
        # Try finding it relative to where script might be if not running from app root
        file_path = Path("resources/competences/FREECAD_COURS.pdf")
        if not file_path.exists():
           # Try absolute path
           file_path = Path("/home/light667/Miabe-Assistant/resources/competences/FREECAD_COURS.pdf")
    
    if not file_path.exists():
        print(f"❌ File not found: {file_path}")
        return

    print(f"🚀 Uploading {file_path}...")
    
    try:
        with open(file_path, "rb") as f:
            file_content = f.read()
            
        storage_path = "competences/FREECAD_Cours.pdf" # Desired path in bucket
        
        supabase.storage.from_("resources").upload(
            storage_path,
            file_content,
            {
                "content-type": "application/pdf",
                "upsert": "true"
            }
        )
        print("✅ Upload successful!")
        
        public_url = supabase.storage.from_("resources").get_public_url(storage_path)
        print(f"🌍 Public URL: {public_url}")
        
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    upload_resource()

File: app/scripts/upload_resources.dart
import 'dart:io';
import 'package:supabase_flutter/supabase_flutter.dart';
// Ensure this matches your project structure

// You can run this script with `dart run scripts/upload_resources.dart`
// Make sure you have the supabase_flutter and dotenv dependencies if needed, 
// or just hardcode keys for this one-off script (but be careful not to commit them if they are secrets).
// For simplicity in this context, we'll ask the user to provide URL/KEY or read from env.

// NOTE: For a dart script outside of Flutter context, we usually use `supabase` package (dart only), 
// but since we are in a Flutter project, we might need to rely on the existing setup or just use standard HTTP if the SDK is too coupled to Flutter.
// However, `supabase_flutter` depends on `supabase` which works in Dart.

void main() async {
  // TODO: Replace with your actual project URL and Anon Key
  const supabaseUrl = 'https://gtnyqqstqfwvncnymptm.supabase.co';
  const supabaseKey = 'YOUR_SUPABASE_ANON_KEY'; // User needs to fill this or we can try to read from a config file

  if (supabaseKey == 'YOUR_SUPABASE_ANON_KEY') {
    print('❌ Error: Please provide your Supabase Anon Key in the script.');
    return;
  }

  final supabase = SupabaseClient(supabaseUrl, supabaseKey);

  const filePath = '../../resources/competences/FREECAD_Cours.pdf';
  final file = File(filePath);

  if (!file.existsSync()) {
    print('❌ Error: File not found at $filePath');
    // Try absolute path if needed or check relative to where script is run
    print('Current directory: ${Directory.current.path}');
    return;
  }

  try {
    print('🚀 Uploading ${file.path}...');
    const fileName = 'competences/FREECAD_Cours.pdf';
    
    // Read file bytes
    final fileBytes = await file.readAsBytes();

    // Upload to 'resources' bucket
    await supabase.storage.from('resources').uploadBinary(
      fileName,
      fileBytes,
      fileOptions: const FileOptions(upsert: true),
    );

    print('✅ Upload successful!');
    
    // Get Public URL
    final publicUrl = supabase.storage.from('resources').getPublicUrl(fileName);
    print('🌍 Public URL: $publicUrl');
    print('Ensure this URL matches what is in home_page.dart');

  } catch (e) {
    print('❌ Error uploading file: $e');
  }
}
